pystac/validation/local_validator.py

```diff
@@ -66,6 +57,15 @@ def item_validator(version: str = VERSION) -> Draft7Validator:
             ] = _read_schema(f"stac-spec/v{version}/{name}.json")
         return Draft7Validator(schema, resolver=resolver)
 
+    def catalog_validator(self, version: str = VERSION) -> Draft7Validator:
```

Comment:
I noticed that these were not figuring out the items sub-schemas properly, so I switched these to just all have everything stored in their resolvers.